src/allmydata/test/cli/test_invite.py

```diff
@@ -102,18 +102,24 @@ def send_messages(wormhole: IWormhole, messages: Sequence[JSONable]) -> None:
 B = TypeVar("B")
 
 def concurrently(
-    client: Callable[[], Awaitable[A]],
-    server: Callable[[], Awaitable[B]],
-) -> defer.Deferred[tuple[A, B]]:
+    client: Callable[[], Union[
+        Coroutine[defer.Deferred[A], Any, A],
+        Generator[defer.Deferred[A], Any, A],
+    ]],
+    server: Callable[[], Union[
+        Coroutine[defer.Deferred[B], Any, B],
+        Generator[defer.Deferred[B], Any, B],
+    ]],
+) -> defer.Deferred[Tuple[A, B]]:
     """
     Run two asynchronous functions concurrently and asynchronously return a
     tuple of both their results.
     """
-    return defer.gatherResults([
+    result = defer.gatherResults([
         defer.Deferred.fromCoroutine(client()),
         defer.Deferred.fromCoroutine(server()),
-    ])
-
+    ]).addCallback(tuple)  # type: ignore
```

Comment:
The mypy error I see here is:

```
src/allmydata/test/cli/test_invite.py:118:34: error: Argument 1 to "gatherResults" has incompatible type "List[object]"; expected
"Iterable[Deferred[<nothing>]]"  [arg-type]
        result = defer.gatherResults([
                                     ^
```

But I can't figure out why.  `client` and `server` obvious return a type that unifies with the element type of the `Iterable` `gatherResults` accepts...

Any ideas?

Reply:
I saw that, but with some tweaking I also saw another one where the issue was that the result is a list, and Python type checking assumes lists have uniform type. Eventually I gave up because meh, it's a test. But maybe worth filing something upstream.

Reply:
Oh, I think it's because of the assumption that list has uniform type, and you're passing list with Deferreds of different types. Filed https://github.com/twisted/twisted/issues/11887